include/ttmlir/Dialect/TTIR/IR/TTIROps.td

```diff
@@ -760,6 +760,29 @@ def TTIR_ConvolutionOp : TTIR_DPSOp<"convolution"> {
   let hasVerifier = 1;
 
   let extraClassDeclaration = [{
+    MutableOperandRange getDpsInitsMutable() { return getOutputMutable(); }
+  }];
+}
+
+def TTIR_GatherOp: TTIR_DPSOp<"gather"> {
+    let summary = "Gather operation.";
+    let description = [{
+      Gather operation.
```

Comment:
I'd love to see better description of this op. GatherOp may have different semantics depending on the source of definition. For example, GatherOp in stable hlo and GatherOp in CCL are completely different. As I can see stable hlo has nice documentation on the op, so we can use some of it.

Reply:
I copied over shlo gather op definition and included reference to shlo.